### DISCUSSION TITLE
EmailStr vs NameEmail

### DISCUSSION THREAD

**Discussion by mblakesley** [pydantic/pydantic#2550] [Question] [answered]
The docs say: (https://pydantic-docs.helpmanual.io/usage/types/#pydantic-types)

> EmailStr
> [...] the input string must be a valid email address, and the output is a simple string
> NameEmail
> [...] the input string must be either a valid email address or in the format Fred Bloggs <fred.bloggs@example.com>. [...]

This suggests `EmailStr` only accepts emails in the form `fred.bloggs@example.com`. But a trivial example shows that `EmailStr` actually accepts either:

```
>>> from pydantic import EmailStr, BaseModel
>>> class User(BaseModel):
...     email: EmailStr
...
...
>>> User(email='a@b.com')
User(email='a@b.com')
>>> User(email='Some text <a@b.com>')
User(email='a@b.com')
>>> User(email='string')  # just to prove validation works
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    User(email='string')
  File "pydantic/main.py", line 362, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for User
email
  value is not a valid email address (type=value_error.email)
```

So what is the actual expected behavior? Basically, is the current behavior wrong or are the docs wrong? Can `EmailStr` be relied upon to accept the format "Fred Bloggs <fred.bloggs@example.com>"?

**Comment by PrettyWood** [comment, marked as answer]
Hi @mblakesley 
By default `pydantic` tries to coerce the input and will hence accept `'Some text <a@b.com>'` for `EmailStr` to keep only the email part. What matters is what you want in your inner model !
```py
from pydantic import BaseModel, EmailStr, NameEmail


class M(BaseModel):
    email: EmailStr
    name_email: NameEmail

m = M(email='Some text <a@b.com>', name_email='Some text <a@b.com>')
assert m.dict() == {
    'email': 'a@b.com',
    'name_email': NameEmail(name='Some text', email='a@b.com')
}
assert M.schema() == {
  "title": "M",
  "type": "object",
  "properties": {
    "email": {
      "title": "Email",
      "type": "string",
      "format": "email"
    },
    "name_email": {
      "title": "Name Email",
      "type": "string",
      "format": "name-email"
    }
  },
  "required": [
    "email",
    "name_email"
  ]
}
```

**Reply by mblakesley** [reply]
@PrettyWood Interesting. Can this behavior be relied upon? Also, shouldn't the docs be updated to say it can accept either form?
